doc/classes/CharacterBody2D.xml

```diff
@@ -98,9 +133,18 @@
 			A higher value means it's more flexible for detecting collision, which helps with consistently detecting walls and floors.
 			A lower value forces the collision algorithm to use more exact detection, so it can be used in cases that specifically require precision, e.g at very low scale to avoid visible jittering, or for stability with a stack of character bodies.
 		</member>
+		<member name="constant_speed_on_floor" type="bool" setter="set_constant_speed_on_floor_enabled" getter="is_constant_speed_on_floor_enabled" default="false">
+			If [code]true[/code], the body will move always at the same speed on the ground no matter the slope. Note you need a use [member floor_snap_strength] for downward slope.
+		</member>
+		<member name="exclude_body_layers" type="int" setter="set_exclude_body_layers" getter="get_exclude_body_layers" default="0">
+			Array of layer that will be excluded from the velocity propagation. If you exclude a collision layer of a moving platform, the  body will not follow it.
+		</member>
 		<member name="floor_max_angle" type="float" setter="set_floor_max_angle" getter="get_floor_max_angle" default="0.785398">
 			Maximum angle (in radians) where a slope is still considered a floor (or a ceiling), rather than a wall, when calling [method move_and_slide]. The default value equals 45 degrees.
 		</member>
+		<member name="floor_snap_strength" type="float" setter="set_floor_snap_strength" getter="get_floor_snap_strength" default="0.0">
+			Strength with which a body will remaining attached to the ground. This will multiply the inverse of the [member up_direction] to determine the snapping vector.
```

Comment:
I only added the first paragraph because the second one is no longer accurate.

As now the `linear_velocity` reflects the real desired velocity, I managed to make `move_and_slide` simpler. Now we just need to call the method.

- It applies the snapping only when the velocity face the inverse of `up_direction`, you can see that in the following condition of `_snap_on_floor()`: `linear_velocity.dot(up_direction) > 0`.
- FYI, it resets the gravity accumulation when you are on ground:
```
	if (on_floor && linear_velocity.dot(up_direction) <= 0) {
		linear_velocity = linear_velocity.slide(up_direction);
	}
```

Reply:
Ah, so you don't need to disable snapping to jump? That's great! I didn't notice that.

I think the description could still explain a little bit more what snapping does, maybe something like that):
```
As long as the snapping vector is in contact with the ground and the body moves against `up_direction`, the body will remain attached to the surface. Snapping is not applied if the body moves along `up_direction`, so it will be able to detach from the ground when jumping.
```